### DISCUSSION TITLE
ROCm Force Flush

### DISCUSSION THREAD

**Discussion by WASasquatch** [comfyanonymous/ComfyUI#250] [Ideas]
In this file
https://github.com/comfyanonymous/ComfyUI/blob/master/execution.py#L198
You mention clearing causes more issues on ROCm, have you tried just filling up the VRAM to force it to flush via exception and just catch and pass? 

basically you call to fill up the vram with dummy tensors of crazy sizes, this clears previous (comfy) related stuck stuff and replaces it, so after it's exception, when comfy tries to run again it should have a clear state. 

This hacky method was needed a lot on colabs and early unoptimized diffusers API.

**Comment by comfyanonymous** [comment]
The problem isn't that it doesn't clear the ram it's that ROCm seems to need to reinit stuff after the vram is cleared. Not clearing the vram on my 6800XT gives me a much smoother experience and I assume it's the same  for all AMD GPUs.

**Reply by WASasquatch** [reply]
Should have left out my RX 480 8GB and RX 580 machines to test this. Though the people handling ROCm are doing a pretty poor job. Why I caved and saved up and got a 4090, actually. Couldn't stand colab, and super limited ability with my AMD cards. Though you at least got 16gb VRAM which is much better then 8gb. Lol
